llumnix/load_computation.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) 2024, Alibaba Group;
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import ABC, abstractmethod
+
+import numpy as np
+
+import llumnix.envs as llumnix_envs
+
+class BaseLoad(ABC):
+    @abstractmethod
+    def is_busy(self) -> bool:
+        """
+        Returns True if the load is busy, False otherwise.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def __lt__(self, other: "BaseLoad") -> bool:
+        """
+        Returns True if the load is less than the other load, False otherwise.
+        """
+        raise NotImplementedError
+
+    @classmethod
+    @abstractmethod
+    def compute_instance_load(cls, instance_info: 'InstanceInfo') -> "BaseLoad":
+        """
+        Compute the load of an instance.
+        """
+        raise NotImplementedError
+
+class DummyLoad(BaseLoad):
+    def __init__(self, *args, **kwargs) -> None:
+        pass
+
+    def is_busy(self) -> bool:
+        return False
+
+    def __lt__(self, other: "DummyLoad") -> bool:
+        return True
+
+    def __repr__(self):
+        return "DummyLoad"
+
+    @classmethod
+    def compute_instance_load(cls, instance_info: 'InstanceInfo') -> "DummyLoad":
+        return DummyLoad()
+
+
+class BlockDemandFactorLoad(BaseLoad):
+    BUSY_THRESHOLD = float(llumnix_envs.BLOCKDEMANDFACTOR_BUSY_THRESHOLD)
```

Comment:
I don't understand what this file is for

Reply:
And it is not put in any sub-directory (llumnix/load_computation.py), looks pretty fundamental...